pkg/asset/imagebased/image/baseiso.go

```diff
@@ -0,0 +1,140 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/coreos/stream-metadata-go/stream"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/rhcos/cache"
+	"github.com/openshift/installer/pkg/types"
+)
+
+var (
+	baseIsoFilename = ""
+
+	defaultCoreOSStreamGetter = rhcos.FetchCoreOSBuild
+)
+
+// BaseIso generates the base ISO file for the image.
+type BaseIso struct {
+	File *asset.File
+
+	streamGetter CoreOSBuildFetcher
+}
+
+// CoreOSBuildFetcher will be to used to switch the source of the coreos metadata.
+type CoreOSBuildFetcher func(ctx context.Context) (*stream.Stream, error)
+
+var _ asset.WritableAsset = (*BaseIso)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BaseIso) Name() string {
+	return "BaseIso Image"
+}
+
+// Dependencies returns dependencies used by the asset.
+func (i *BaseIso) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the base ISO.
+func (i *BaseIso) Generate(dependencies asset.Parents) error {
+	var err error
+	var baseIsoFileName string
```

Comment:
You have a global var with the same name.

Reply:
That's an agent leftover :see_no_evil: I believe this global var is supposed to be used in `Load`, but it is never set anyway and `Load` ends up being just some non-reachable code. This just doesn't lead to any weird behavior/failure, because in `Generate`, we are anyway using the [RHCOS cache](https://github.com/openshift/installer/blob/master/pkg/rhcos/cache/cache.go#L269-L293), which trying to load the base ISO from the cache directory. So `Load` and this global var are redundant.

I'll verify this and update this PR (then I'll propose a minor fix for the agent codebase). Thanks for pointing this out :+1: